groups/bsl/bslstl/bslstl_optional.t.cpp

```diff
@@ -395,7 +613,8 @@ struct MyClass2 {
         else {
```

Comment:
I copy pasted these classes from bslma_constructionutil so all the decisions are inherited from there. I'll modify as suggested

Reply:
from email conversation "I think that it would make sense to make the test types "well behaved" - even if it doesn't matter now, it might be confusing in the future.  Let's do the minimal possible amount of changes though - in this case it's just the move constructor I believe."
Done